ui/app/models/oidc/assignment.js

```diff
@@ -1,14 +1,62 @@
 import Model, { attr, hasMany } from '@ember-data/model';
+import ArrayProxy from '@ember/array/proxy';
+import PromiseProxyMixin from '@ember/object/promise-proxy-mixin';
 import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import { isPresent } from '@ember/utils';
 
+const validations = {
+  name: [
+    { type: 'presence', message: 'Name is required.' },
+    // ARG TODO add in after Claire pushes her branch
+    // {
+    //   type: 'containsWhiteSpace',
+    //   message: 'Name cannot contain whitespace.',
+    // },
+  ],
+  targets: [
+    {
+      validator(model) {
+        const entityIds = model.hasMany('entity_ids').ids();
+        const groupIds = model.hasMany('group_ids').ids();
+        return isPresent(entityIds) || isPresent(groupIds);
+      },
+      message: 'At least one entity or group is required.',
+    },
+  ],
+};
+
+@withModelValidations(validations)
 export default class OidcAssignmentModel extends Model {
   @attr('string') name;
-  @hasMany('identity/entity') entityIds;
-  @hasMany('identity/group') groupIds;
+  @hasMany('identity/entity') entity_ids;
+  @hasMany('identity/group') group_ids;
 
   @lazyCapabilities(apiPath`identity/oidc/assignment/${'name'}`, 'name') assignmentPath;
   @lazyCapabilities(apiPath`identity/oidc/assignment`) assignmentsPath;
 
+  get targets() {
+    return ArrayProxy.extend(PromiseProxyMixin).create({
+      promise: this.prepareTargets(),
+    });
+  }
+
+  async prepareTargets() {
+    const targets = [];
+
+    for (const key of ['entity_ids', 'group_ids']) {
+      (await this[key]).forEach((model) => {
```

Comment:
Something to consider: what happens here if there are none of one of the models (I believe that's a 404 from the API) or they don't have permissions (403)?